docs/root/configuration/http/http_filters/proto_message_extraction_filter.rst

```diff
@@ -0,0 +1,106 @@
+.. _config_http_filters_proto_message_extraction:
+
+Proto Message Extraction
+========================
+
+* gRPC :ref:`architecture overview <arch_overview_grpc>`
+* This filter should be configured with the type URL ``type.googleapis.com/envoy.extensions.filters.http.proto_message_extraction.v3.ProtoMessageExtractionConfig``.
+* :ref:`v3 API reference <envoy_v3_api_msg_extensions.filters.http.proto_message_extraction.v3.ProtoMessageExtractionConfig>`.
+
+ProtoMessageExtraction filter supports extracting gRPC
+requests/responses (proto messages) into google.protobuf.Struct and storing
+results in the dynamic metadata `envoy.filters.http.proto_message_extraction`
+for later access.
+
+Use Case
+--------
+
+The ProtoMessageExtraction filter is particularly useful in scenarios where
+sensitive or detailed logging of gRPC requests and responses is required.
+In Client-Side Streaming or Server-Side Streaming, the filter can store the
+first and last messages, which can later be used for logging or for obtaining
+a comprehensive view of the data flow.
+
+Assumptions
+-----------
+
+This filter assumes it is only applicable for gRPC with Protobuf as payload.
+
+Process Flow
+------------
+
+On the request and response path, it will check
+
+1. if the incoming gRPC request/response is configured, the filter tries to:
```

Comment:
Could you document if the extraction is at critical path?

Reply:
Sure, I have updated the file regarding the same. Please let me know if anything needs clarification.